app/cdap/components/PipelineDetails/store/ActionCreator.js

```diff
@@ -296,6 +296,25 @@ const getStatistics = (params) => {
   );
 };
 
+const getMetadataEndpoints = ({ namespace, appId, workflowType, runId }) => {
```

Comment:
should be setMetadataEndpoints?

Reply:
I followed the general naming pattern in the `ActionCreator`. I believe the `get` is due to them getting the response from the backend API. `getStatistics`, `getNextRuntime`, etc. do similar stuff.

Reply:
ur i see those. I still don't think it quite make sense, because ultimately you are dispatching the SET, even a name like `getThenSet` makes more sense to me. But this is fine as it it.
And you and remove the `return`, since you are not actually using the subscription elsewhere